docs(sync): document escaping `#` as `%23` in sync paths (#7193)
juicedata/juicefs: docs/en/guide/sync.md

You can sync data between any [supported storage system](../reference/how_to_set_up_object_storage.md), but note that if one of the endpoint is a JuiceFS volume, it it then recommended to [sync without mount point](#sync-without-mount-point) since it runs without FUSE overhead.

:::note Paths containing special characters

> `juicefs sync` parses both the source and destination paths as URLs, so if a path contains characters that have a special meaning in a URL (such as `#`), they must be escaped (URL-encoded) first, otherwise the path will be truncated or parsed incorrectly. For example, `#` must be written as `%23`, so the path `a/b#c/d` should be written as `a/b%23c/d`.

Reviewer: ```suggestion
`juicefs sync` interprets both source and destination paths as URLs. Therefore, if a path contains characters with special meanings in URLs (such as `#`), they must be escaped (URL-encoded) before use. Otherwise, the path may be truncated or parsed incorrectly. For example, `#` must be written as `%23`, so the path `a/b#c/d` should be written as `a/b%23c/d`.
```
